src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -56,21 +57,30 @@ static zns_sm_metadata_t make_metadata(
   
   auto config_segment_size = get_conf<Option::size_t>(
     "seastore_segment_size");
-  INFO("config segment size: {}", config_segment_size);
-  size_t zones_per_segment = config_segment_size / zone_capacity;
+
+  size_t segment_size = (zone_size == 0) ? config_segment_size : (zone_size << SECT_SHIFT);
```

Comment:
Why the segment_size has to be zone_size << 9 instead of the configured size?

Reply:
zone_size is reported as sectors(of 512 bytes) from the ioctl. That same value is being passed on to make_metadata().
For example, for a zonesize of 64MB, ioctl returns 131072 as zone size, to convert it to bytes we need to multiply it with 512 or do a left shift by 9. 
I am realising that it would be better to convert zone-size from sectors to bytes and then write it to metadata.
Will do that in next update.